### DISCUSSION TITLE
Next Release(1.1.2)

### DISCUSSION THREAD

**Discussion by whimet** [performancecopilot/parfait#116] [General]
@natoscott Security vulnerabilities from [Dependabot](https://github.com/performancecopilot/parfait/security/dependabot) have been cleared, do you think it's ready to create the next release?

**Comment by natoscott** [comment]
@whimet great work - go right ahead  ( if @tallpsmith is around he might be able to double-check and assist )

**Comment by tallpsmith** [comment]
If everyone believes a new Parfait release is ready to spin, I _think_ I can still do that.  Just let me know the release # you want to release it under and I'll give it a crack once you've confirmed.  I'll just trust @natoscott  and @whimet that it's ready 😅

**Comment by whimet** [comment]
Here're the list of production code changes since release 1.1.1. @natoscott If you have no problem of the commits between 22218a3..8c0270c, I'd like to propose a 1.1.2 release.

* 91d4b2a - Bump xstream in parfait-cxf from 1.4.19 to 1.4.20 (5 days ago) (Yanhui Li)
* 5eac445 - Migrate parfait-cxf from cxf-api:2.2.3 to cxf-core 4.0.3 (11 days ago) (Yanhui Li)
* 8e65095 - Bump jackson-core/databind/annotations version from 2.11.2 to 2.16.0 (12 days ago) (Yanhui Li)
* 0516acf - Bump jetty from 6.1.20 to 6.1.23 in /parfait-cxf (1 year, 6 months ago) (dependabot[bot])
* 20ce040 - Bump spring-expression in /parfait-cxf (9 months ago) (dependabot[bot])
* c29c7a5 - Bump spring-core from 5.3.18 to 5.3.27 (8 months ago) (dependabot[bot])
* 449d120 - Bump guava from 30.0-jre to 32.1.3-jre in parfait-core/parfait-dxm/parfait-dropwizard (4 weeks ago) (Yanhui Li)
* 8c0270c - Bump spring-core from 4.3.28.RELEASE to 5.3.18 (1 year, 8 months ago) (dependabot[bot])
* a91c7f3 - Fix warnings from deprecated methods (1 year, 9 months ago) (Ryan Doyle)
* 4cd6502 - Bump xstream from 1.4.18 to 1.4.19 in /parfait-cxf (1 year, 10 months ago) (dependabot[bot])
* f6b6eeb - drop log4j config files (1 year, 10 months ago) (Andreas Gerstmayr)
* 60ec194 - Drop ThreadContext test which can no longer be implemented (1 year, 10 months ago) (Nathan Scott)
* 3565c04 - Tweak json definition of runtime_name string to use units=none (2 years ago) (Nathan Scott)
* 55cdfcf - Transition parfait agent and core to using slf4j exclusively (2 years ago) (Nathan Scott)
* 6e41573 - Update parfait-agent metric definitions for now-optional GC metrics (2 years ago) (Nathan Scott)
* c315cd5 - Update guava version Parfait uses to avoid CVE-2020-8908 (2 years ago) (Nathan Scott)
* 22218a3 - Bump xstream from 1.4.16 to 1.4.18 in /parfait-cxf (2 years, 3 months ago) (dependabot[bot])

**Reply by natoscott** [reply]
@whimet LGTM, go right ahead.

**Reply by whimet** [reply]
Thanks @natoscott, LGTM as well. @tallpsmith Can you please create a 1.1.2 release?

**Reply by tallpsmith** [reply]
I'm going to try and release today if I can.

**Reply by tallpsmith** [reply]
@whimet Just want you to be aware that these changes will require Parfait to have a minimum of Java 17:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project parfait-cxf: Compilation failure
[ERROR] /Users/psmith/dev/personal/parfait/parfait-cxf/src/main/java/io/pcp/parfait/cxf/MonitoringInterceptor.java:[19,34] cannot access org.apache.cxf.interceptor.Fault
[ERROR]   bad class file: /Users/psmith/.m2/repository/org/apache/cxf/cxf-core/4.0.3/cxf-core-4.0.3.jar(org/apache/cxf/interceptor/Fault.class)
[ERROR]     class file has wrong version 55.0, should be 52.0
[ERROR]     Please remove or make sure it appears in the correct subdirectory of the classpath.
```

(If I try to use the default Java 8 we've been using)

**Reply by whimet** [reply]
@tallpsmith Hmm, I missed it, sorry for that.

Just wondering, does this error occur during the release process? As the CI build passed on Java 8.

**Reply by tallpsmith** [reply]
The Github workflow _attempted_ to install Java 8, but in reality the _default_ Java 11 was being used.  So, technically, the minimum is Java 11, but if we're going to uprev things I thought it made more sense to lift it up to Java 17 if we have to move.

Is this going to be a problem for Babylon? Is that still running Java8?

**Reply by tallpsmith** [reply]
I should probably release as `1.2.0` given the change in underlying JDK requirements, just trying to confirm if I can get it passing with Java 11 cleanly and will make that a minimum.  But I'm very certain it was never actually passing under Java8.

**Reply by whimet** [reply]
@tallpsmith I agree that Java 17 makes more sense with a new release, and 1.2.0 sounds good.

Babylon is currently on Java 11, and will probably upgrade to 17 in 2024.

**Comment by tallpsmith** [comment]
I’ve managed to bring it back to Java 11 so that should help.

**Reply by tallpsmith** [reply]
@whimet `1.2.0` is now released to the Public OSS Sonatype Maven repos and should be referenceable.  This supports Java 11+ now, so hopefully that works for you.

Thanks for your assistance upgrading things.

**Reply by whimet** [reply]
Great, thanks a lot @tallpsmith!
